File: app/src/main/java/org/carstenf/wordfinder/LetterButton.java
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder;

import android.widget.Button;

import org.jetbrains.annotations.NotNull;

class LetterButton {

	final private int pos;
	final private Button button;

	LetterButton(int pos, @NotNull Button button) {
		this.pos=pos;
		this.button=button;
	}

	public void setText(String string) {
		button.setText(string);
	}

	private boolean pressed = false;
	void setEnabled(boolean b) {
		pressed = b;
		button.setPressed(b);
	}

	boolean isEnabled() {
		return pressed;
	}

	int getPos() {
		return pos;
	}

	public void setContentDescription(String s) {
		button.setContentDescription(s);
	}
}

File: app/src/main/kotlin/org/carstenf/wordfinder/LetterButton.kt
/*
 * Copyright Carsten Friedrich (Carsten.Friedrich@gmail.com)
 *
 * License: GNU GENERAL PUBLIC LICENSE 3.0 (https://www.gnu.org/copyleft/gpl.html)
 *
 */
package org.carstenf.wordfinder

internal class LetterButton(val pos: Int, private val button: AppCompatLetterButton) {
    fun setText(string: String?) {
        button.text = string
    }

    private var checked = false

    var isChecked: Boolean
        get() = checked
        set(b) {
            checked = b
            button.isChecked = !b
        }

    fun setContentDescription(s: String?) {
        button.contentDescription = s
    }
}
